### PR TITLE
Feature/better-multicall

### PR DESCRIPTION
**Pull request by janek26**
### Issue / feature description

Allows for an user to upgrade his account to a different account implementation.

### Changes

- add new screen to pick implementation
- rename setting "use plugin account" to "show implementation picker"
- make upgrade logic more robust

### Checklist

- [x] Rebased to the last commit of the target branch (or merged)
- [x] Code self-reviewed
- [x] Code self-tested
- [x] Tests updated (if needed)
- [x] All tests are passing locally

Please keep your pull request as small as possible. If you need to make multiple changes, please create separate pull requests for each change. Create a draft pull request if you need early feedback. This will mark the pull request as a work in progress and prevent it from being reviewed or merged until you are ready.

Please move drafts to the ready for review (regular PR) when you are ready to start the review process and other developers will pick it up from there.

### CODE REVIEW

File: packages/extension/src/background/wallet.ts [hunk 1]
```diff
@@ -201,13 +202,11 @@ export class Wallet {
     accountType: ArgentAccountType,
   ): Promise<string> {
```
Comment (diegodelrieu) [inline packages/extension/src/background/wallet.ts:203]: type can be updated to `Promise<keyof Required<Network>["accountClassHash"]>` I guess no ? Or TS might be able to infer it

File: packages/extension/src/shared/network/type.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ export interface Network {
   accountClassHash?: {
     argentAccount: string
     argentPluginAccount?: string
+    argentBetterMulticallAccount?: string
```
Comment (diegodelrieu) [inline packages/extension/src/shared/network/type.ts:13]: why better?

File: packages/extension/src/ui/components/Menu.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import styled from "styled-components"
```
Comment (diegodelrieu) [inline packages/extension/src/ui/components/Menu.tsx:1]: Is there a particular reason to mix up Chakra and styled components ?

File: packages/extension/src/ui/services/addresses.ts [hunk 4]
```diff
@@ -69,8 +69,11 @@ export const addressSchema = yup
 export const isValidAddress = (address: string) =>
   addressSchema.isValidSync(address)
 
-export const isEqualAddress = (a: string, b: string) => {
+export const isEqualAddress = (a: string, b?: string) => {
```
Comment (diegodelrieu) [inline packages/extension/src/ui/services/addresses.ts:72]: isnt this worth creating a new util ? it seems like it's serving a slightly different purpose here

File: packages/extension/src/shared/network/type.ts (hunk 2 above)
Comment (janek26) [reply packages/extension/src/shared/network/type.ts:13]: https://community.starknet.io/t/proposal-to-improve-multicall-allowing-to-chain-transactions/2038

File: packages/extension/src/ui/services/addresses.ts (hunk 4 above)
Comment (janek26) [reply packages/extension/src/ui/services/addresses.ts:72]: I think having it here is just fine, and undefined can not be equal to the address in string, so I think it keeps the functionality it promises

File: packages/extension/src/background/wallet.ts (hunk 1 above)
Comment (janek26) [reply packages/extension/src/background/wallet.ts:203]: I did not touch that code
While I think it's a good improvement, it has implications in the code where this function is used that I dont want to mess with right now for this feature

File: packages/extension/src/ui/features/accountEdit/AccountImplementationScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        {otherImplementations && (
+          <>
+            <H3>Other implementations</H3>
+            <AutoColumn gap="md">
+              {otherImplementations.map((i) => (
+                <ImplementationItem
+                  {...i}
+                  key={i.id}
+                  active={false}
+                  onClick={async () => {
```
[the remark this answers is not in the document]
Comment (janek26) [reply packages/extension/src/ui/features/accountEdit/AccountImplementationScreen.tsx]: thanks :)

File: packages/extension/src/ui/components/Menu.tsx (hunk 3 above)
Comment (janek26) [reply packages/extension/src/ui/components/Menu.tsx:1]: no - we used styled-components before and are transitioning to chakra.

I just took existing code and moved it to be able to delete a much bigger file :)

File: packages/extension/src/ui/services/addresses.ts (hunk 4 above)
Comment (dhruvkelawala) [reply packages/extension/src/ui/services/addresses.ts:72]: Do you think we can do this?

https://github.com/argentlabs/argent-x/blob/ea9b1c1eb412c7417de1ad0cbaed2efd4084abd8/packages/extension/src/ui/services/addresses.ts#L72

**Review by dhruvkelawala — APPROVED** [review]
lgtm

**Pull request merged**